R/get.R

```diff
@@ -278,16 +278,22 @@ get_unapproved_use <- function(approved_packages, used_packages) {
 #'
 #' @param file File path of file being run
 #'
-#' @importFrom lintr lint
-#'
 #' @return results from `lintr::lint()`
 #'
 #' @noRd
 #'
 get_lint_results <- function(file) {
+
+   if (!requireNamespace("lintr", quietly = TRUE)) {
+      message(strwrap("Linting will not be included in the log. Install the
+         lintr package to use the log.rx.lint feature.",
+         prefix = " ", initial = ""))
+      return()
+   }
+
    # lint file if option is turned on
    if (!is.logical(getOption('log.rx.lint'))) {
-      lint(file, getOption('log.rx.lint'))
+      lintr::lint(file, getOption('log.rx.lint'))
```

Comment:
do we need `::` if we have the function imported?

Reply:
I'm okay with leaving them in - we just haven't done a lot of `::` in the past.  I'm alway unclear what is the best practice.

Reply:
Here was my thought process for using `::`.  We no longer import the function to remove the dependency on `lintr`.  So instead we warn users if they don't have `lintr` installed, using `requireNamespaces()` to check, and use `lintr::` instead of attaching the package with `require()` to avoid unexpected changes to a users search path.

Reply:
ah that makes sense!

Reply:
For packages that are not depends/imports, you need to use the package::function notation. @nicholas-masel is right that you wouldn't want to use require since it changes the environment (loading the package into the search path). 

I also suggest naming the arguments just to make sure nothing gets mixed up.

```suggestion
      lintr::lint(filename = file, linters = getOption('log.rx.lint'))
```